src/extension/neoExpress/packageVersion.ts

```diff
@@ -0,0 +1,73 @@
+import tryFetchJson from "../util/tryFetchJson";
+
+export default class PackageVersion {
+  major: number;
+  minor: number;
+  patch: number;
+  private allLatestVersions: PackageVersion[] = [];
+
+  constructor(major: number, minor: number, patch: number) {
+    this.major = major;
+    this.minor = minor;
+    this.patch = patch;
+  }
+
+  static parse(version: string): PackageVersion {
+    const parts = version.split(".");
+    if (parts.length !== 3) {
+      throw new Error(`Invalid version string: ${version}`);
+    }
+    return new PackageVersion(parseInt(parts[0]), parseInt(parts[1]), parseInt(parts[2]));
+  }
+
+  async findLatestPatchVersionFromNuget(): Promise<PackageVersion> {
+    return (
+      (await this.getLatestVersionsFromNuget())
+        .filter((target) => this.compare(target) < 0 && !this.isNewMajorOrMinorVersion(target))
+        .pop() || this
+    );
+  }
+
+  toString() {
+    return `${this.major}.${this.minor}.${this.patch}`;
+  }
+
+  equals(other: PackageVersion, ignorePatchVersion: boolean = false) {
+    if (ignorePatchVersion) {
+      return this.compare(other) === 0;
+    }
+    return this.major === other.major && this.minor === other.minor;
+  }
+
+  isNewMajorOrMinorVersion(other: PackageVersion) {
+    return this.compare(other) < 0 && (this.major !== other.major || this.minor !== other.minor);
+  }
+
+  compare(other: PackageVersion) {
+    if (this.major > other.major) {
+      return 1;
+    } else if (this.major < other.major) {
+      return -1;
+    } else if (this.minor > other.minor) {
+      return 1;
+    } else if (this.minor < other.minor) {
+      return -1;
+    } else if (this.patch > other.patch) {
+      return 1;
+    } else if (this.patch < other.patch) {
+      return -1;
+    }
+    return 0;
+  }
+
+  private async getLatestVersionsFromNuget(): Promise<PackageVersion[]> {
+    if (this.allLatestVersions.length > 0) {
+      return this.allLatestVersions;
+    }
+    let response = await tryFetchJson("https", "api.nuget.org", "/v3-flatcontainer/neo.express/index.json");
```

Comment:
Can we have a configuration setting to use the [build server feed](https://dev.azure.com/ngdenterprise/Build/_artifacts/feed/public)? That feed URL is `https://pkgs.dev.azure.com/ngdenterprise/c96908c2-e4b5-4c77-b955-4b690f24380b/_packaging/9e84eb49-63f0-4b48-a8c4-039901073643/nuget/v3/flat2/neo.express/index.json`

Reply:
Note, if we do that, we might want to first fetch the `v3/index.json` file from the nuget feed root and then find the `PackageBaseAddress/3.0.0` resource in that file to find the `flatcontainer` base URI

Reply:
Maybe I can introduce a configuration setting that will default to the official nuget feed. Users can customize it. This way, we provide flexibility and don't need to be very smart in the code.